pdns/pdnsutil.cc

```diff
@@ -641,6 +644,11 @@ static int checkZone(DNSSECKeeper &dk, UeberBackend &B, const DNSName& zone, con
         continue;
       }
       if( rr.qname.isPartOf( qname.first ) ) {
+        if( qname.second == QType::DNAME && rr.qname == qname.first ) {
```

Comment:
can't say these if/if/if/if blocks are very readable anymore. took me quite some time to understand why this is a new case over the first if. i don't understand however why this is inside the rr.qname.isPartOf bit.

Reply:
I see now that 3 lines up also mentions DNAME. I can probably do better.

Reply:
combining the ifs would not have made things more readable. Instead I added a bunch of comments.